src/status_im/ui/screens/wallet/main/views.cljs

```diff
@@ -104,54 +106,44 @@
       [react/text {:style styles/add-asset-text}
        (i18n/label :t/wallet-add-asset)]]]]])
 
-(defn render-asset [{:keys [id currency amount]}]
-  ;; TODO(jeluard) Navigate to asset details screen
-  #_
-  [list/touchable-item show-not-implemented!
-   [react/view
-    [list/item
-     [list/item-image {:uri :launch_logo}]
-     [react/view {:style styles/asset-item-value-container}
-      [react/text {:style styles/asset-item-value} (str amount)]
-      [react/text {:style      styles/asset-item-currency
-                   :uppercase? true}
-       id]]
-     [list/item-icon {:icon :icons/forward}]]]]
-  (if id
-    [react/view
-     [list/item
-      (let [{:keys [source style]} (token->image id)]
-        [list/item-image source style])
-      [react/view {:style styles/asset-item-value-container}
-       [react/text {:style           styles/asset-item-value
-                    :number-of-lines 1
-                    :ellipsize-mode  :tail}
-        (money/to-fixed (money/wei-> :eth amount))]
-       [react/text {:style           styles/asset-item-currency
-                    :uppercase?      true
-                    :number-of-lines 1}
-        id]]]]
+(defn render-asset [{:keys [name symbol decimals amount] :as m}]
+  (if name ;; If no 'name' then this the dummy value used to render `add-asset`
+    [list/touchable-item #(utils/show-popup "TODO" (str "Details about " symbol " here"))
+     [react/view
+      [list/item
+       (let [{:keys [source style]} (token->icon m)]
+         [list/item-image source style])
+       [react/view {:style styles/asset-item-value-container}
+        [react/text {:style           styles/asset-item-value
+                     :number-of-lines 1
+                     :ellipsize-mode  :tail}
+         (money/to-fixed (money/token->unit (or amount 0) decimals))]
+        [react/text {:style           styles/asset-item-currency
+                     :uppercase?      true
+                     :number-of-lines 1}
+         symbol]]
+       [list/item-icon {:icon :icons/forward}]]]]
     [add-asset]))
 
-(defn asset-section [balance prices-loading? balance-loading?]
-  (let [assets (concat [{:id "eth" :currency :eth :amount balance}]
-                       (if config/erc20-enabled?
-                         [{:id "snt" :currency :snt :amount 5000000000000000000000}]))]
+(def ethereum-asset {:name "Ethereum" :symbol :ETH :decimals 18})
+
+(defn tokens-for [network]
+  (get tokens/all (ethereum/network network)))
+
+(defn asset-section [network balance prices-loading? balance-loading?]
+  (let [tokens (tokens-for network)
+        assets (map #(assoc % :amount (get balance (:symbol %))) (concat [ethereum-asset] tokens))]
```

Comment:
This does not check for the flag any more. If it's disabled, it should give you just ETH like before.

Reply:
I don't see much value for the flag in this case. The point of this PR is to add new feature so it's not like before :)
Next I will work on assets default so a user can decide what tokens are visible by default.